File: Python/Program/PowerSupply.py
from PowerDevice import PowerDevice

class PowerSupply(PowerDevice):
    def __init__(self, daya=0, sertifikasi="", modular=False, wattage=0, efficiency_rating="", kecepatan_baca=0, kecepatan_tulis=0):
        super().__init__(wattage, efficiency_rating, kecepatan_baca, kecepatan_tulis)
        self.daya = daya
        self.sertifikasi = sertifikasi
        self.modular = modular
    
    def set_daya(self, daya):
        self.daya = daya
    
    def set_sertifikasi(self, sertifikasi):
        self.sertifikasi = sertifikasi
    
    def set_modular(self, modular):
        self.modular = modular
    
    def get_daya(self):
        return self.daya
    
    def get_sertifikasi(self):
        return self.sertifikasi
    
    def get_modular(self):
        return self.modular
    
    def is_modular(self):
        return self.modular
    
    def turn_on(self):
        print("Power supply turned on.")
    
    def turn_off(self):
        print("Power supply turned off.")
    
    def get_info(self):
        self.get_power_info()
        print(f"Daya: {self.daya} W, Sertifikasi: {self.sertifikasi}, Modular: {'Ya' if self.modular else 'Tidak'}")

File: Java/Program/PowerSupply.java
public class PowerSupply extends PowerDevice {
    private int daya;
    private String sertifikasi;
    private boolean modular;

    public PowerSupply() {
    }

    public PowerSupply(int daya, String sertifikasi, boolean modular, int wattage, String efficiencyRating, int kecepatanBaca, int kecepatanTulis) {
        super(wattage, efficiencyRating, kecepatanBaca, kecepatanTulis);
        this.daya = daya;
        this.sertifikasi = sertifikasi;
        this.modular = modular;
    }

    public void setDaya(int daya) {
        this.daya = daya;
    }

    public void setSertifikasi(String sertifikasi) {
        this.sertifikasi = sertifikasi;
    }

    public void setModular(boolean modular) {
        this.modular = modular;
    }

    public int getDaya() {
        return daya;
    }

    public String getSertifikasi() {
        return sertifikasi;
    }

    public boolean isModular() {
        return modular;
    }

    public void turnOn() {
        System.out.println("Power supply turned on.");
    }

    public void turnOff() {
        System.out.println("Power supply turned off.");
    }

    public void getInfo() {
        getPowerInfo();
        System.out.println("Daya: " + daya + " W, Sertifikasi: " + sertifikasi + ", Modular: " + (modular ? "Ya" : "Tidak"));
    }
}